tests/test_generator.py

```diff
@@ -0,0 +1,88 @@
+"""Test generator class"""
+import unittest
+from unittest.mock import patch, mock_open, MagicMock
+from pathlib import Path
+import pandas as pd
+from aind_data_schema_models._generators.generator import generate_code
+
+
+class TestGenerateCode(unittest.TestCase):
+    """Test generate_code"""
+    @patch("builtins.open", new_callable=mock_open, read_data="template content")
+    @patch("pandas.read_csv")
+    @patch("subprocess.run")
+    @patch("jinja2.Environment.from_string")
+    def test_generate_code(self, mock_from_string, mock_subprocess_run, mock_read_csv, mock_open):
+        """Test the generate_code function"""
+        # Mock the CSV data to be used
+        mock_data = pd.DataFrame({"column": ["value"]})
+        mock_read_csv.return_value = mock_data
+
+        # Mock Jinja2 template rendering
+        mock_template = MagicMock()
+        mock_template.render.return_value = "rendered code"
+        mock_from_string.return_value = mock_template
+
+        # Define the paths that will be used in the function
+        data_type = "test_data"
+        output_path = Path("./src/aind_data_schema_models/test_data.py")
+
+        # Run the function with isort and black enabled
+        generate_code(data_type, isort=True, black=True)
+
+        # Check if the CSV file was read correctly
+        mock_read_csv.assert_called_once_with(Path(f"./src/aind_data_schema_models/_generators/models/{data_type}.csv"))
```

Comment:
Sometimes it's convenient to run the tests individually. You can set an absolute path to this file like it's done here:
https://github.com/AllenNeuralDynamics/aind-data-schema-models/blob/main/tests/test_utils.py
which makes it easy to run the tests from different directories and different operating systems.

Reply:
added